utilities/transactions/write_unprepared_transaction_test.cc

```diff
@@ -188,6 +188,96 @@ TEST_P(WriteUnpreparedTransactionTest, ReadYourOwnWrite) {
   delete txn;
 }
 
+TEST_P(WriteUnpreparedTransactionTest, RecoveryRollbackUnprepared) {
+  WriteOptions write_options;
+  write_options.disableWAL = false;
+  uint64_t seq_used = kMaxSequenceNumber;
+  uint64_t log_number;
+  WriteBatch batch;
+  std::vector<Transaction*> prepared_trans;
+  WriteUnpreparedTxnDB* wup_db;
+  options.disable_auto_compactions = true;
+
+  // Try unprepared batches with empty database.
+  for (int num_batches = 0; num_batches < 10; num_batches++) {
+    // Reset database.
+    prepared_trans.clear();
+    ReOpen();
+    wup_db = dynamic_cast<WriteUnpreparedTxnDB*>(db);
+
+    // Write num_batches unprepared batches into the WAL.
+    for (int i = 0; i < num_batches; i++) {
+      batch.Clear();
+      ASSERT_OK(WriteBatchInternal::InsertNoop(&batch));
```

Comment:
Where do we call MarkBeginPrepare? How is that logic tested?

Reply:
Can you add a TODO to replace this manual construction of the write batch with a call to txn->Write (when that part of the code is done)? It would be hard to read/maintain the low level batch construction code if it is not centralized in one class.